odl/solvers/scalar/steplen.py

```diff
@@ -148,6 +148,21 @@ def __call__(self, x, direction, dir_derivative):
             The computed step length
         """
         alpha = 1.0
+
+        # Since all comparisons with nan returns False, in order to not get a
+        # problem with nan-values we throw an error.
+        if self.function(x) == float('nan'):
+            raise NameError('ObjFunIsNanError')
```

Comment:
For now I think it can raise a ```ValueError```. But it might be worth thinking about what can be done instead...